lib/trino-plugin-toolkit/src/main/java/io/trino/plugin/base/group/CachingGroupProviderConfig.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.base.group;
+
+import io.airlift.configuration.Config;
+import io.airlift.configuration.ConfigDescription;
+import io.airlift.units.Duration;
+import jakarta.validation.constraints.Min;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+public class CachingGroupProviderConfig
+{
+    private Duration ttl = new Duration(5, SECONDS);
+    private long cacheMaximumSize = Long.MAX_VALUE;
```

Comment:
what about caching missing entries?

Reply:
The `CachingGroupProvider` will cache whatever the delegate returns, so as far as it's concerned, there are no "missing entries". OTOH, some caches have separate configuration for "negative results", and sometimes it makes sense to skip caching those, but I don't think it makes sens to make this distinction here.